### ISSUE TITLE
Exception: ModuleNotFoundError: No module named 'poetry.core'

### ISSUE THREAD

**Issue by bparzella** [sinoroc/tox-poetry-dev-dependencies#5]
When I install and use the package, I get the following error:

```
Traceback (most recent call last):
  File "C:\Users\USER\.pyenv\pyenv-win\versions\3.6.8\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\USER\.pyenv\pyenv-win\versions\3.6.8\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "d:\Data\.venv\Scripts\tox.exe\__main__.py", line 9, in <module>
  File "d:\Data\.venv\lib\site-packages\tox\session\__init__.py", line 44, in cmdline
    main(args)
  File "d:\Data\.venv\lib\site-packages\tox\session\__init__.py", line 65, in main
    config = load_config(args)
  File "d:\Data\.venv\lib\site-packages\tox\session\__init__.py", line 81, in load_config
    config = parseconfig(args)
  File "d:\Data\.venv\lib\site-packages\tox\config\__init__.py", line 263, in parseconfig
    pm = get_plugin_manager(plugins)
  File "d:\Data\.venv\lib\site-packages\tox\config\__init__.py", line 79, in get_plugin_manager
    pm.load_setuptools_entrypoints("tox")
  File "d:\Data\.venv\lib\site-packages\pluggy\manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "d:\Data\.venv\lib\site-packages\importlib_metadata\__init__.py", line 95, in load
    module = import_module(match.group('module'))
  File "C:\Users\USER\.pyenv\pyenv-win\versions\3.6.8\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "d:\Data\.venv\lib\site-packages\tox_poetry_dev_dependencies\_hooks.py", line 10, in <module>
    import poetry.core.factory
ModuleNotFoundError: No module named 'poetry.core'
Running tests failed, please check tox output.
```

After manually installing the package python-core tox runs as expected. Should poetry-core be a requirement of tox-poetry-dev-dependencies?

I'm using poetry 1.0.10 on windows with python 3.6.

**Comment by sinoroc** [comment]
Ah, right, I forgot about that. I did something that is probably a bit silly...

In short: right now, one should install `tox-poetry-dev-dependencies[plugin_tox]`.

For the long explanation... So yes, [_poetry-core_](https://pypi.org/project/poetry-core/) is a dependency of this plugin. But this project being a _Tox_ plugin also means that it defines an _entry-point_ to let Tox know where to find the hooks. See [Tox's documentation section "_Tutorial: a minimal tox plugin_"](https://tox.readthedocs.io/en/3.20.0/plugins.html#tutorial-a-minimal-tox-plugin). And on the other hand, reading [_setuptools_ documentation section "_Dynamic Discovery of Services and Plugins_"](https://setuptools.readthedocs.io/en/latest/setuptools.html#dynamic-discovery-of-services-and-plugins) one could interpret that, on the plugin-side the dependencies should be declared as optional. And it is the responsibility of the calling tool (in our case Tox) to check if those dependencies are present before calling the plugin hooks. Looks like Tox doesn't do this check, but the end result is more or less the same: the plugin doesn't work.

I think probably in this case it doesn't make much sense to set the dependency on _poetry-core_ as optional. Since the plugin has strictly no functionality on its own and can only be useful as a plugin of Tox. This feature of having extra dependencies for some entry points was probably intended for projects that can be used both independently and as a plugin of another. And the entry points are only active when the two projects are present in the same environment.

So I will think it over, and will probably make the dependency a mandatory one. In the meantime, one should either install `poetry-core` in an additional step or install with the right _extra_: `tox-poetry-dev-dependencies[plugin_tox]`.

Thanks for the report.

**Comment by sinoroc** [comment]
Should be fixed in [version 0.0.1](https://pypi.org/project/tox-poetry-dev-dependencies/0.0.1/).

**Comment by bparzella** [comment]
Nice, thank you!
